c/cert/src/rules/CON34-C/AppropriateThreadObjectStorageDurations.ql

```diff
@@ -0,0 +1,46 @@
+/**
+ * @id c/cert/appropriate-thread-object-storage-durations
+ * @name CON34-C: Declare objects shared between threads with appropriate storage durations
+ * @description Accessing thread-local variables with automatic storage durations can lead to
+ *              unpredictable program behavior.
+ * @kind problem
+ * @precision high
+ * @problem.severity error
+ * @tags external/cert/id/con34-c
+ *       correctness
+ *       concurrency
+ *       external/cert/obligation/rule
+ */
+
+import cpp
+import codingstandards.c.cert
+import codingstandards.cpp.Concurrency
+import semmle.code.cpp.dataflow.TaintTracking
+import semmle.code.cpp.dataflow.DataFlow
+
+class MallocFunctionCall extends FunctionCall {
```

Comment:
is malloc the only acceptable dynamic memory allocation? do we accept new as well? if so maybe use [this instead](https://github.com/github/codeql/blob/main/cpp/ql/lib/semmle/code/cpp/commons/Alloc.qll#L25)?

Reply:
`realloc` is probably acceptable -- I can add a test case!

Reply:
Thanks for finding that library function -- I was digging around for something like that and couldn't find it.